plugins/outputs/application_insights/application_insights.go

```diff
@@ -0,0 +1,427 @@
+	var telemetryValue float64
+	err := setFloat64TelemetryProperty(&telemetryValue, []string{fieldName}, metric, nil)
+	if err != nil {
+		return nil
+	}
+
+	var telemetryName string
+	if useFieldNameInTelemetryName {
+		telemetryName = metric.Name() + "_" + fieldName
+	} else {
+		telemetryName = metric.Name()
+	}
+	telemetry := appinsights.NewMetricTelemetry(telemetryName, telemetryValue)
+	telemetry.Properties = metric.Tags()
+	a.addContextTags(metric, telemetry)
+	telemetry.Timestamp = metric.Time()
+	return telemetry
+}
+
+func (a *ApplicationInsights) createAggregateMetricTelemetry(metric telegraf.Metric) (*appinsights.AggregateMetricTelemetry, []string) {
+	var telemetryValue float64
+	var telemetryCount int
+	usedFields := make([]string, 0, 6) // We will use up to 6 fields
+	var err error
+
+	// Get the sum of all individual measurements(mandatory property)
+	err = setFloat64TelemetryProperty(&telemetryValue, []string{"sum", "value"}, metric, &usedFields)
+	if err != nil {
+		return nil, nil
+	}
+
+	// Get the count of measurements (mandatory property)
+	err = setIntTelemetryProperty(&telemetryCount, []string{"count", "samples"}, metric, &usedFields)
+	if err != nil {
+		return nil, nil
+	}
+
+	telemetry := appinsights.NewAggregateMetricTelemetry(metric.Name())
+	telemetry.Value = telemetryValue
+	telemetry.Count = telemetryCount
+	telemetry.Properties = metric.Tags()
+	a.addContextTags(metric, telemetry)
+	telemetry.Timestamp = metric.Time()
+
+	// We attempt to set min, max, variance and stddev fields but do not really care if they are not present--
+	// they are not essential for aggregate metric.
+	// By convention AppInsights prefers stddev over variance, so to be consistent, we test for stddev after testing for variance.
+	setFloat64TelemetryProperty(&(telemetry.Min), []string{"min"}, metric, &usedFields)
+	setFloat64TelemetryProperty(&(telemetry.Max), []string{"max"}, metric, &usedFields)
+	setFloat64TelemetryProperty(&(telemetry.Variance), []string{"variance"}, metric, &usedFields)
+	setFloat64TelemetryProperty(&(telemetry.StdDev), []string{"stddev"}, metric, &usedFields)
+
+	return telemetry, usedFields
+}
+
+func (a *ApplicationInsights) createTelemetryForUnusedFields(metric telegraf.Metric, usedFields []string) []appinsights.Telemetry {
+	fields := metric.Fields()
+	retval := make([]appinsights.Telemetry, 0, len(fields))
+
+	for fieldName := range fields {
+		if contains(usedFields, fieldName) {
+			continue
+		}
+
+		telemetry := a.createSimpleMetricTelemetry(metric, fieldName, true)
+		if telemetry != nil {
+			retval = append(retval, telemetry)
+		}
+	}
+
+	return retval
+}
+
+func (a *ApplicationInsights) addContextTags(metric telegraf.Metric, telemetry appinsights.Telemetry) {
+	for tagSourceName, contextTagName := range a.ContextTagSources {
+		contextTagValue, found := metric.GetTag(tagSourceName)
```

Comment:
I saw you use style like below before, which looks pretty concise
if contextTagValue, found := metric.GetTag(tagSourceName); found {
      telemetry.ContextTags()[contextTagName] = contextTagValue 
}

Reply:
True, but in this particular case this idiom cannot be used because I need the contextTagValue two lines below

Reply:
The contextTagValue is used inside the scope?

Reply:
Oh, I see what you mean. Will fix, thx